deployment/sriov-network-operator/templates/certificate.yaml

```diff
@@ -0,0 +1,72 @@
+{{- if .Values.operator.admissionControllers.enabled }}
+{{- if and (.Values.operator.admissionControllers.certificates.certManager.enabled) (.Values.operator.admissionControllers.certificates.certManager.generateSelfSigned) }}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ .Values.operator.admissionControllers.certificates.secretNames.operator }}
+  namespace: {{ .Namespace }}
+spec:
+  dnsNames:
+  - operator-webhook-service.{{ .Namespace }}.svc
+  - operator-webhook-service.{{ .Namespace }}.svc.cluster.local
+  issuerRef:
+    kind: Issuer
+    name: operator-webhook-selfsigned-issuer
+  secretName: {{ .Values.operator.admissionControllers.certificates.secretNames.operator }}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: operator-webhook-selfsigned-issuer
+  namespace: {{ .Namespace }}
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ .Values.operator.admissionControllers.certificates.secretNames.injector }}
+  namespace: {{ .Namespace }}
+spec:
+  dnsNames:
+  - network-resources-injector-service.{{ .Namespace }}.svc
+  - network-resources-injector-service.{{ .Namespace }}.svc.cluster.local
+  issuerRef:
+    kind: Issuer
+    name: network-resources-injector-selfsigned-issuer
+  secretName: {{ .Values.operator.admissionControllers.certificates.secretNames.injector }}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: network-resources-injector-selfsigned-issuer
+  namespace: {{ .Namespace }}
+spec:
+  selfSigned: {}
+{{- end }}
+{{- if and (not .Values.operator.admissionControllers.certificates.certManager.enabled) (.Values.operator.admissionControllers.certificates.custom.enabled) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Values.operator.admissionControllers.certificates.secretNames.operator }}
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+  ca.crt: {{ .Values.operator.admissionControllers.certificates.custom.operator.caCrt | b64enc | b64enc | quote }}
```

Comment:
double b64enc is needed?

Reply:
Yes. We mount the `ca.crt` field of the secret as an environment variable. When you do that, under the hood it will decode once and set it as env variable on pod's env. Then the business logic (app) gets this one-time-encoded string from env and pushes it as is to the [caBundle](https://github.com/kubernetes/api/blob/12b5cfde901cc29a42bf09ae64307818f6cfd874/admissionregistration/v1/types.go#L584-L587) which [expects base64 encoded content](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#:~:text=Note%3A%20You%20must%20replace%20the%20%3CCA_BUNDLE%3E%20in%20the%20above%20example%20by%20a%20valid%20CA%20bundle%20which%20is%20a%20PEM%2Dencoded%20(field%20value%20is%20Base64%20encoded)%20CA%20bundle%20for%20validating%20the%20webhook%27s%20server%20certificate.). 

Double encoding helps also with consuming the secret as env variable instead of trying to consume a value that looks like this (with which I actually got some errors and then I went with the double base64 encoding idea):
```
-----BEGIN CERTIFICATE-----
MIIMIICLDCCAdKgAwIBAgIBADAKBggqhkjOPQQDAjB9MQswCQYDVQQGEwJCRTEPMA0G
...
-----END CERTIFICATE-----
```